tests/core/fs/workspacefs/test_workspace_fs.py

```diff
@@ -438,3 +441,77 @@ async def mockup(*args):
     with running_backend.offline():
         with pytest.raises(FSBackendOfflineError):
             await alice_workspace.get_reencryption_need()
+
+
+@pytest.mark.trio
+async def test_upload_blocks(alice_user_fs, alice2_user_fs, running_backend, monkeypatch):
+    wid = await create_shared_workspace("w", alice_user_fs, alice2_user_fs)
+    alice_workspace = alice_user_fs.get_workspace(wid)
+
+    alice2_workspace = alice2_user_fs.get_workspace(wid)
+
+    fspath = "/taz"
+
+    await alice_workspace.touch(fspath)
+    await alice_workspace.sync()
+
+    await alice_workspace.write_bytes(fspath, b"aaaaaa" * DEFAULT_BLOCK_SIZE)
+
+    info = await alice_workspace.path_info(fspath)
+
+    assert info["base_version"] == 1
+    assert info["is_placeholder"] is False
+    assert info["need_sync"] is True
+
+    assert await alice2_workspace.read_bytes(fspath) == bytearray()
+
+    # mock the synchronization_step method to retrieve the new manifest
+    vanilla_transaction_synchronization_step = alice_workspace.transactions.synchronization_step
```

Comment:
Can't we just use a block counter in the mock_bcakend_block_create to know when to raise an error without having to also mock the synchronization step ?

```python
blocks_create_before_crash = TAR_V2_BLOCKS // 2  # Crash in the middle of the upload

async def mock_backend_block_create(*args, **kwargs):
    nonlocal blocks_create_before_crash
    if blocks_create_before_crash == 0:
        raise BlockNotFoundError()
    blocks_create_before_crash -= 1
    return await vanilla_backend_block_create(*args, **kwargs)

monkeypatch.setattr(running_backend.backend.block, "create", mock_backend_block_create)
```

Reply:
I used a similar logic in the first version of my test, but I removed it because with a counter, if we had several files in the workspace, we are not sure on wich file the error will be raised.

Reply:
Not a probleme, in this new version, we use a new workspace.